aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -549,12 +609,26 @@ module aptos_framework::fungible_asset {
     fun withdraw_internal(
         store_addr: address,
         amount: u64,
-    ): FungibleAsset acquires FungibleStore {
+    ): FungibleAsset acquires FungibleStore, ConcurrentFungibleBalance {
         assert!(amount != 0, error::invalid_argument(EAMOUNT_CANNOT_BE_ZERO));
-        let store = borrow_global_mut<FungibleStore>(store_addr);
-        assert!(store.balance >= amount, error::invalid_argument(EINSUFFICIENT_BALANCE));
-        store.balance = store.balance - amount;
 
+        if (auto_upgrade_to_concurrent_fungible_balance()) {
+            ensure_store_upgraded_to_concurrent_internal(store_addr);
+        };
+
+        if (concurrent_fungible_balance_exists(store_addr)) {
```

Comment:
why are we happy pathing aggregator

Reply:
with making it opt-in , updated to happy path non-aggregator